src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -0,0 +1,413 @@
+    unsigned int shard_groups_count = m_shards.size() / num_DS_clusters;
+    if ((m_shards.size() % num_DS_clusters) > 0)
+    {
+        shard_groups_count++;
+    }
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "DEBUG num of shard group count " << shard_groups_count)
+
+    my_DS_cluster_num = m_consensusMyID / DS_MULTICAST_CLUSTER_SIZE;
+    my_shards_lo = my_DS_cluster_num * shard_groups_count;
+    my_shards_hi = my_shards_lo + shard_groups_count - 1;
+
+    if (my_shards_hi >= m_shards.size())
+    {
+        my_shards_hi = m_shards.size() - 1;
+    }
+}
+
+void DirectoryService::SendVCBlockToShardNodes(
+    unsigned int my_DS_cluster_num, unsigned int my_shards_lo,
+    unsigned int my_shards_hi, vector<unsigned char>& vcblock_message)
+{
+    // Too few target shards - avoid asking all DS clusters to send
+    LOG_MARKER();
+
+    if ((my_DS_cluster_num + 1) <= m_shards.size())
+    {
+        auto p = m_shards.begin();
+        advance(p, my_shards_lo);
+
+        for (unsigned int i = my_shards_lo; i <= my_shards_hi; i++)
+        {
+            vector<Peer> shard_peers;
+
+            for (auto& kv : *p)
+            {
+                shard_peers.push_back(kv.second);
+                LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                          " PubKey: "
+                              << DataConversion::SerializableToHexStr(kv.first)
+                              << " IP: " << kv.second.GetPrintableIPAddress()
+                              << " Port: " << kv.second.m_listenPortHost);
+            }
+
+            P2PComm::GetInstance().SendBroadcastMessage(shard_peers,
+                                                        vcblock_message);
+            p++;
+        }
+    }
+}
+
+void DirectoryService::ProcessViewChangeConsensusWhenDone()
+{
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "View change consensus is DONE!!!");
+
+    m_pendingVCBlock->SetCoSignatures(*m_consensusObject);
+
+    unsigned int index = 0;
+    unsigned int count = 0;
+
+    vector<PubKey> keys;
+    for (auto& kv : m_mediator.m_DSCommitteePubKeys)
+    {
+        if (m_pendingVCBlock->GetB2().at(index) == true)
+        {
+            keys.push_back(kv);
+            count++;
+        }
+        index++;
+    }
+
+    // Verify cosig against vcblock
+    shared_ptr<PubKey> aggregatedKey = MultiSig::AggregatePubKeys(keys);
+    if (aggregatedKey == nullptr)
+    {
+        LOG_GENERAL(WARNING, "Aggregated key generation failed");
+    }
+
+    vector<unsigned char> message;
+    m_pendingVCBlock->GetHeader().Serialize(message, 0);
+    m_pendingVCBlock->GetCS1().Serialize(message, VCBlockHeader::SIZE);
+    BitVector::SetBitVector(message, VCBlockHeader::SIZE + BLOCK_SIG_SIZE,
+                            m_pendingVCBlock->GetB1());
+    if (Schnorr::GetInstance().Verify(message, 0, message.size(),
```

Comment:
Understand that `== false` makes this explict, but how about using `not` keyword http://en.cppreference.com/w/cpp/language/operator_alternative